erpnext/stock/doctype/material_request/material_request.py

```diff
@@ -442,19 +472,29 @@ def select_item(d):
 		{
 			"Material Request": {
 				"doctype": "Purchase Order",
-				"validation": {"docstatus": ["=", 1], "material_request_type": ["=", "Purchase"]},
+				"validation": {
+					"docstatus": ["=", 1],
+					"material_request_type": ["in", ["Purchase", "Subcontracting"]],
+				},
 			},
 			"Material Request Item": {
 				"doctype": "Purchase Order Item",
 				"field_map": [
-					["name", "material_request_item"],
-					["parent", "material_request"],
-					["uom", "stock_uom"],
-					["uom", "uom"],
-					["sales_order", "sales_order"],
-					["sales_order_item", "sales_order_item"],
-					["wip_composite_asset", "wip_composite_asset"],
-				],
+					item
+					for item in [
+						["name", "material_request_item"],
+						["parent", "material_request"],
+						["qty", "fg_item_qty"] if is_subcontracted else [],
+						["item_code", "fg_item"] if is_subcontracted else [],
+						["uom", "stock_uom"] if not is_subcontracted else [],
+						["uom", "uom"] if not is_subcontracted else [],
+						["sales_order", "sales_order"],
+						["sales_order_item", "sales_order_item"],
+						["wip_composite_asset", "wip_composite_asset"],
+					]
+					if item
```

Comment:
Instead of list comprehension, add it in the separate function and return the list of fields to mapped from that function